src/main/groovy/life/qbic/datamodel/datasets/datastructure/files/nanopore/DriftCorrectionLog.groovy

```diff
@@ -0,0 +1,32 @@
+package life.qbic.datamodel.datasets.datastructure.files.nanopore
+
+import life.qbic.datamodel.datasets.datastructure.files.DataFile
+
+/**
+ * A specialisation of a DataFile, represents an Oxford Nanopore Drift Correction file
+ *
+ * @author: Sven Fillinger
+ */
+class DriftCorrectionLog extends DataFile {
+
+    final private static String FILE_TYPE = "csv"
+
+    final private static String NAME_SCHEMA = $/drift_correction_.*/$
+
+    protected DriftCorrectionLog() {}
+
+    protected DriftCorrectionLog(String name, String relativePath) {
+        super(name, relativePath, FILE_TYPE)
+        validateName()
+    }
```

Comment:
```suggestion
    protected DriftCorrectionLog(String name, String relativePath, String fileType) {
        if (fileType != FILE_TYPE) {
            throw new IllegalArgumentException("${DriftCorrectionLog.class} file type not correct")
        }
        super(name, relativePath, fileType)
        validateName()
    }
```
The fileType is completely ignored and set by this class.

Reply:
no, the filetype is an intrinsic property of the Drift Correction Log, it is a special case of a data file, where the file type is preset.

Reply:
We decided to set the static field FILE_TYPE and validate the parent class using the super() constructor. The create method does not take any fileType argument.